subprojects/docs/src/docs/userguide/authoringMaintainableBuildScripts.adoc

```diff
@@ -0,0 +1,112 @@
+// Copyright 2018 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[authoring_maintainable_build_scripts]]
+== Authoring Maintainable Build Scripts
+
+Gradle build scripts combine the qualities of declarative build logic, expressiveness as well as flexibility and rigidity as needed. As a build script author it is easy to fall into the trap of striking the wrong balance or applying poor coding habits. This chapter describes best practices for writing your build script in a meaningful, yet flexible and efficient way.
+
+[NOTE]
+====
+The third-party link:https://github.com/nebula-plugins/gradle-lint-plugin[Gradle lint plugin] helps with enforcing a desired code style in a build script if you are looking for appropriate linting automation.
+====
+
+[[sec:declare_task_group_and_description]]
+=== Declare the group and description for tasks
```

Comment:
I wonder whether this really ought to be the first tip. I was hoping for something "cooler".

Reply:
Can you expand on what you'd expect to be "cooler"? In terms of the writing, the content or the features? I moved this to the first position as it is the easiest to approach and apply for users.

Reply:
Oh, I understand your comment now. You were hoping for a better first _topic_ within the chapter. I don't think I have a strong opinion here.